### PR TITLE
User research preparation

### PR DESCRIPTION
**Pull request by camilleldn**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,11 +27,15 @@ gem 'mail'
...
 gem 'therubyracer', platforms: :ruby
 
+gem 'unicorn'
+
 group :development, :test do
   gem 'byebug'
   gem 'pry-rails'
   gem 'rspec-rails'
   gem 'rubocop'
+  gem 'sqlite3'
```
Comment (gjvis) [inline Gemfile]: Problems running postgres?

File: app/controllers/occupations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class OccupationsController < ApplicationController
+  def update
+    report = Report.find_for_guid(params[:report_id])
```
Comment (gjvis) [inline app/controllers/occupations_controller.rb]: `Report.friendly.find(params[:report_id])`

File: app/controllers/occupations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    occupation = report.find_occupation(params[:id].to_i)
```
Comment (gjvis) [inline app/controllers/occupations_controller.rb]: is the `to_i` neeed?

File: app/controllers/occupations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  private
+
+  def occupation_accepted?(params)
+    params[:commit].downcase != 'refuse'
+  end
+
+  def action_params
+     params.select {|key, value| ["actions", "notes"].include?(key) }
```
Comment (gjvis) [inline app/controllers/occupations_controller.rb]: Rails 4 has [built in support](http://guides.rubyonrails.org/action_controller_overview.html#strong-parameters) for this, like so:
```ruby
params.permit(:actions, :notes)
```
or, if they're required params (this blows up if any required params are missing though):
```ruby
params.require(:actions, :notes)
```

File: app/models/occupation.rb [hunk 5]
```diff
@@ -8,4 +8,26 @@ class Occupation < ActiveRecord::Base
...
            :additional_titles,
            to: :soc_occupation
+  has_many :actions
+
+  def formatted_tasks
+    tasks.split(';').map(&:strip).map(&:capitalize)
+  end
+
+  def mark_as_refused
+    self.accepted = false
```
Comment (gjvis) [inline app/models/occupation.rb]: `update(accepted: false)`

File: Gemfile (hunk 1 above)
Comment (camilleldn) [inline Gemfile]: ayup :) turns out sudo access would be really quite helpful at nices ;) fixing

File: app/models/report.rb [hunk 6]
```diff
@@ -19,16 +18,54 @@ def self.generate_report_for_keywords(search_keywords)
     )
   end
 
+  def self.find_for_guid(guid)
```
Comment (gjvis) [inline app/models/report.rb]: don't need this if we use `Foo.friendly.find()`

File: app/models/report.rb [hunk 7]
```diff
@@ -19,16 +18,54 @@ def self.generate_report_for_keywords(search_keywords)
...
+    where(guid: guid).first
+  end
+
+  def find_occupation(occupation_id)
+    occupations.find { |o| o.id == occupation_id }
```
Comment (gjvis) [inline app/models/report.rb]: `occupations.find(:id)`

File: app/models/report.rb [hunk 8]
```diff
@@ -19,16 +18,54 @@ def self.generate_report_for_keywords(search_keywords)
...
+    occupations.select(&:accepted)
+  end
+
   def selected_occupations
-    occupations.where(selected: true)
+    occupations.select(&:selected)
+  end
+
+  def complete?
+    selected_occupations.count > 0 && !occupations_to_review?
```
Comment (gjvis) [inline app/models/report.rb]: ```selected_occupations.any? && !occupations_to_review?```

**Pull request merged**